src/test/java/uk/gov/hmcts/reform/pip/account/management/service/AccountServiceTest.java

```diff
@@ -44,138 +45,149 @@
+        assertEquals(0, createdAccounts.get(CreationEnum.CREATED_ACCOUNTS).size(),
                      "Map should have no created accounts"
         );
     }
 
     @Test
     void testValidationCreatesAnErroredAccount() {
-        Subscriber subscriber = new Subscriber();
-        subscriber.setEmail(EMAIL);
+        AzureAccount azureAccount = new AzureAccount();
+        azureAccount.setEmail(EMAIL);
 
-        when(validator.validate(argThat(sub -> ((Subscriber) sub).getEmail().equals(subscriber.getEmail()))))
+        when(validator.validate(argThat(sub -> ((AzureAccount) sub).getEmail().equals(azureAccount.getEmail()))))
             .thenReturn(Set.of(constraintViolation));
 
         when(constraintViolation.getMessage()).thenReturn(VALIDATION_MESSAGE);
+        when(constraintViolation.getPropertyPath()).thenReturn(path);
+        when(path.toString()).thenReturn(EMAIL_PATH);
 
-        Map<CreationEnum, List<? extends Subscriber>> createdSubscribers =
-            accountService.createSubscribers(List.of(subscriber));
+        Map<CreationEnum, List<? extends AzureAccount>> createdAccounts =
+            accountService.addAzureAccounts(List.of(azureAccount), ISSUER_EMAIL);
 
-        assertTrue(createdSubscribers.containsKey(CreationEnum.ERRORED_ACCOUNTS), ERRORED_ACCOUNTS_VALIDATION_MESSAGE);
-        List<? extends Subscriber> subscribers = createdSubscribers.get(CreationEnum.ERRORED_ACCOUNTS);
-        assertEquals(subscriber.getEmail(), subscribers.get(0).getEmail(), EMAIL_VALIDATION_MESSAGE);
-        assertNull(subscriber.getAzureSubscriberId(), "Subscriber should have no azure ID set");
+        assertTrue(createdAccounts.containsKey(CreationEnum.ERRORED_ACCOUNTS), ERRORED_ACCOUNTS_VALIDATION_MESSAGE);
+        List<? extends AzureAccount> accounts = createdAccounts.get(CreationEnum.ERRORED_ACCOUNTS);
+        assertEquals(azureAccount.getEmail(), accounts.get(0).getEmail(), EMAIL_VALIDATION_MESSAGE);
+        assertNull(azureAccount.getAzureAccountId(), "Account should have no azure ID set");
 
-        assertEquals(VALIDATION_MESSAGE, ((ErroredSubscriber) subscribers.get(0)).getErrorMessages().get(0),
-                     "Subscriber should have error message set when validation has failed"
+        assertEquals(EMAIL_PATH + ": " + VALIDATION_MESSAGE,
+                     ((ErroredAzureAccount) accounts.get(0)).getErrorMessages().get(0),
+                     "Account should have error message set when validation has failed"
         );
 
-        assertEquals(0, createdSubscribers.get(CreationEnum.CREATED_ACCOUNTS).size(),
+        assertEquals(0, createdAccounts.get(CreationEnum.CREATED_ACCOUNTS).size(),
                      "Map should have no created accounts"
         );
     }
 
     @Test
-    void creationOfMultipleSubscribers() throws AzureCustomException {
-        Subscriber subscriber = new Subscriber();
-        subscriber.setEmail(EMAIL);
+    void creationOfMultipleAccounts() throws AzureCustomException {
```

Comment:
these tests are pretty huge, can we strip some out? the expected user seems like its the same as the one above so could maybe be extracted, along with instantiating an `AzureAccount` in a before each and then individual fields needed for change to be modified at test level? along with any repeated mockito whens?

Reply:
Updated :) I've moved the duplicate when's / accounts up to the top